compiler/semtempl.nim

```diff
@@ -616,10 +616,10 @@ proc semTemplateDef(c: PContext, n: PNode): PNode =
   result = n
   var s: PSym
   if isTopLevel(c):
-    s = semIdentVis(c, skTemplate, n[namePos], {sfExported})
+    s = semIdentVis(c, skTemplate, n[namePos], {sfExported}, true)
     incl(s.flags, sfGlobal)
   else:
-    s = semIdentVis(c, skTemplate, n[namePos], {})
+    s = semIdentVis(c, skTemplate, n[namePos], {}, true)
```

Comment:
Instead of the refactoring, why not add code here like

```nim

    let info = getLineInfo(n[namePos])
    suggestSym(c.graph, info, s, c.graph.usageSym)

```

Reply:
The "refactoring" I referred to was moving the getLineInfo proc from semstmts.nim (where it was internal to semIdentDef) into sem.nim, so it could be used from elsewhere. Sorry, my commit comment should have been more clear.

The rationale for calling getLineInfo from semIdentVis in semtypes.nim was future-proofing - if in the future there is another "identifier with visibility" definition that needs to be picked up by nimsuggest, then the code to do that is already there. But you would know much better than I about whether such a thing is at all likely.

On the other hand, your suggestion reduces the scope of the change, which is certainly a good thing. If you prefer that option please let me know and I will update the PR accordingly.

Reply:
>  If you prefer that option please let me know and I will update the PR accordingly.

Yes, please do as I outlined. Procs growing ever more switches is not "future proof".